### PR TITLE
fix: remove possible old reports "before" building node

### PR DESCRIPTION
**Pull request by laurgarn**
US 1120

### CODE REVIEW

**Review by SlimaneAmar — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/study/server/StudyService.java [hunk 1]
```diff
@@ -1950,6 +1950,12 @@ private BuildInfos getBuildInfos(UUID nodeUuid) {
     public void buildNode(@NonNull UUID studyUuid, @NonNull UUID nodeUuid) {
         BuildInfos buildInfos = getBuildInfos(nodeUuid);
         updateBuildStatus(nodeUuid, BuildStatus.BUILDING);
+        try {
+            buildInfos.getReportUuids().forEach(reportService::deleteReport);
+        } catch (HttpClientErrorException.NotFound e) {
```
Comment (SlimaneAmar) [inline src/main/java/org/gridsuite/study/server/StudyService.java:1955]: Remove try-catch

Comment (laurgarn) [reply src/main/java/org/gridsuite/study/server/StudyService.java:1955]: OK, as the deleteReport request has ERROR_ON_REPORT_NOT_FOUND, false.

File: src/test/java/org/gridsuite/study/server/StudyTest.java [hunk 2]
```diff
@@ -529,6 +533,12 @@ public MockResponse dispatch(RecordedRequest request) {
                                 .setBody(mapper.writeValueAsString(lineAttachResponseInfos))
                                 .addHeader("Content-Type", "application/json; charset=utf-8");
                     }
+                } else if ("DELETE".equals(request.getMethod()) && path.contains("reports")) {
```
Comment (niconoir) [inline src/test/java/org/gridsuite/study/server/StudyTest.java:536]: I do not understand what you are trying to achieve here.

Comment (laurgarn) [reply src/test/java/org/gridsuite/study/server/StudyTest.java:536]: It was to please Coverage about the catch (*NOT_FOUND) I added around the deleteReport call

File: src/test/java/org/gridsuite/study/server/StudyTest.java [hunk 3]
```diff
@@ -3410,8 +3423,11 @@ private void testBuildFailedWithNodeUuid(UUID studyUuid, UUID nodeUuid) throws E
         assertEquals(nodeUuid, buildStatusMessage.getHeaders().get(HEADER_NODE));
         assertEquals(UPDATE_TYPE_BUILD_FAILED, buildStatusMessage.getHeaders().get(HEADER_UPDATE_TYPE));
 
-        assertTrue(getRequestsDone(1).stream()
-                .anyMatch(r -> r.matches("/v1/networks/" + NETWORK_UUID_2_STRING + "/build\\?receiver=.*")));
+        String requestPath;
+        do {
+            requestPath = getRequestsDone(1).iterator().next();
+        } while (requestPath.contains("reports"));
```
Comment (FranckLecuyer) [inline src/test/java/org/gridsuite/study/server/StudyTest.java:3429]: Same as above

File: src/test/java/org/gridsuite/study/server/StudyTest.java (hunk 2 above)
Comment (niconoir) [reply src/test/java/org/gridsuite/study/server/StudyTest.java:536]: So you can remove it when you will remove the `catch (HttpClientErrorException.NotFound e)` (see @SlimaneAmar comment) ?

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

File: src/test/java/org/gridsuite/study/server/StudyTest.java [hunk 4]
```diff
@@ -3361,8 +3371,11 @@ private void testBuildWithNodeUuid(UUID studyUuid, UUID nodeUuid) throws Excepti
         assertEquals(UPDATE_TYPE_BUILD_COMPLETED, buildStatusMessage.getHeaders().get(HEADER_UPDATE_TYPE));
         assertEquals(Set.of("s1", "s2"), buildStatusMessage.getHeaders().get(HEADER_UPDATE_TYPE_SUBSTATIONS_IDS));
 
-        assertTrue(getRequestsDone(1).stream()
-                .anyMatch(r -> r.matches("/v1/networks/" + NETWORK_UUID_STRING + "/build\\?receiver=.*")));
+        String requestPath;
+        do {
+            requestPath = getRequestsDone(1).iterator().next();
+        } while (requestPath.contains("reports"));
```
[the remark this answers is not in the document]
Comment (SlimaneAmar) [reply src/test/java/org/gridsuite/study/server/StudyTest.java:3377]: We need to test the exact number expected in a unit test

File: src/test/java/org/gridsuite/study/server/StudyTest.java [hunk 5]
```diff
@@ -3361,8 +3361,11 @@ private void testBuildWithNodeUuid(UUID studyUuid, UUID nodeUuid) throws Excepti
         assertEquals(UPDATE_TYPE_BUILD_COMPLETED, buildStatusMessage.getHeaders().get(HEADER_UPDATE_TYPE));
         assertEquals(Set.of("s1", "s2"), buildStatusMessage.getHeaders().get(HEADER_UPDATE_TYPE_SUBSTATIONS_IDS));
 
-        assertTrue(getRequestsDone(1).stream()
-                .anyMatch(r -> r.matches("/v1/networks/" + NETWORK_UUID_STRING + "/build\\?receiver=.*")));
+        IntStream.range(0, nbReportExpected).forEach(i -> {
+            assertTrue(getRequestsDone(1).iterator().next().contains("reports"));
+        });
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/study/server/StudyTest.java:3366]: assertTrue(getRequestsDone(nbReportExpected).stream().allMatch(r -> r.contains("reports")));

File: src/test/java/org/gridsuite/study/server/StudyTest.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (laurgarn) [reply src/test/java/org/gridsuite/study/server/StudyTest.java]: Done

File: src/test/java/org/gridsuite/study/server/StudyTest.java (hunk 3 above)
Comment (laurgarn) [reply src/test/java/org/gridsuite/study/server/StudyTest.java]: Done

File: src/test/java/org/gridsuite/study/server/StudyTest.java (hunk 2 above)
Comment (laurgarn) [reply src/test/java/org/gridsuite/study/server/StudyTest.java]: Done

File: src/main/java/org/gridsuite/study/server/StudyService.java (hunk 1 above)
Comment (laurgarn) [reply src/main/java/org/gridsuite/study/server/StudyService.java]: Done

File: src/test/java/org/gridsuite/study/server/StudyTest.java [hunk 6]
```diff
@@ -3361,8 +3361,11 @@ private void testBuildWithNodeUuid(UUID studyUuid, UUID nodeUuid) throws Excepti
...
+        assertTrue(getRequestsDone(1).iterator().next()
```
Comment (SlimaneAmar) [inline src/test/java/org/gridsuite/study/server/StudyTest.java:3367]: Why do you change this line ?

**Review by FranckLecuyer — APPROVED** [review]

**Review by SlimaneAmar — APPROVED** [review]

**Pull request merged**